### ISSUE TITLE
memcache ttl too short on kilo -> liberty for running rpc-support play

### ISSUE THREAD

**Issue by b3rnard0** [rcbops/rpc-openstack#967]
##### What version (commit) do you encounter this issue on?
V12.1.0-rc1

##### What type of issue is this (bug, feature, documentation, regression)?
bug,regression

##### What is the expected behavior?
play succeeds

##### What is actually happening?
play fails

##### How can we reproduce this issue?
make rpc-support play take a long time

##### How can we verify your issue is addressed?
never hit missing rpc_support ssh key fetching from memcache

##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?
rerun the play

##### What is your evaluation on the importance of this issue, and why?
medium/low?

###### Remove branches (lines) you do not wish to target.
- [x] Master #977 
- [x] liberty-12.1 #978
- [x] liberty-12.0 #1010 
- [x] kilo #1014 

@sigmavirus24 can you confirm this?

**Comment by BjoernT** [comment]
The default TTL should be 24h so I'm not sure if that is correct, the object could have been evicted too.
Personally I would prefer to get rid of memcache a distribution and spend the extra time to implement this via ansible facts

**Comment by prometheanfire** [comment]
nope :D

https://github.com/openstack/openstack-ansible-plugins/blob/master/library/memcached#L87

**Comment by BjoernT** [comment]
Lol, of course our driver alters the default ttl :-)

**Comment by txgunslinger77** [comment]
We seem to be running into this problem again/still in r12.0.1, with the rpc-support play failing with memcache errors, so we still need someone to look at this please. We get these quite often with very little idea as to the solution or work aorund.

**Comment by prometheanfire** [comment]
ok, reopening as a request to backport to the 12.0 series

**Comment by b3rnard0** [comment]
@git-harry If you can make the eval that this is a backport candidate for r12.0.2, it would be much appreciated.

**Comment by txgunslinger77** [comment]
For a little more context, I'm running deploy.sh and get the following error at the following task (for brevity, I only included one sample error, there are many more just like it):

TASK: [rpc_support | Retrieve authorized keys] ********************************
failed: [768021-compute003] => (item={'src': '/root/.ssh/rpc_support', 'name': 'rpc_support', 'file_mode': '0600', 'dir_mode': '0755'}) => {"err": "Memcache key not found", "failed": true, "item": {"dir_mode": "0755", "file_mode": "0600", "name": "rpc_support", "src": "/root/.ssh/rpc_support"}, "rc": 1}
msg: The key you specified was not found within memcached. If you are load balancing memcached, attempt to connect to a single node.

**Comment by BjoernT** [comment]
The change would literally like changing 300 to 86400 inside the rpcd/playbooks/roles/rpc_support/tasks/distribute_auth_key.yml play so from what I am concerned, we NEED to backport it. It's failing too often

**Comment by prometheanfire** [comment]
@BjoernT it was backported, are you saying it needs to be backported further?

**Comment by sigmavirus24** [comment]
@BjoernT #1010 backported this to 12.0.x (probably 12.0.2 at this point) and based on #977 and #988 this will be in both 12.1 and 12.2. Where else does this need to be if those are not satisfactory?

**Comment by BjoernT** [comment]
We would need it inside the kilo branch too since we still add hosts on that code base

**Comment by sigmavirus24** [comment]
@BjoernT you say "It's failing too often" which (when combined with the context of your last comment) implies to me that you've seen this before on Kilo but never reported it. Is that the case?

**Comment by BjoernT** [comment]
Yes correct, it was for quite some time only sporadic but now it seems to increase which I think is related to the prolonged run time due to the larger installations we have and the key is expiring in the meantime

**Comment by sigmavirus24** [comment]
Re-opening per @BjoernT's comments so @git-harry can make a go/no-go call on backporting this to kilo

**Comment by git-harry** [comment]
I see no issue with backporting this. I will target the issue at the next patch milestone for 11.1 so it is tracked.

**Comment by txgunslinger77** [comment]
UPDATE:
First - I manually added the TTL line to the file and no issues I know of were experienced, so it's probably a good idea to include it.

Second - My sincere apologies for any time spent chasing this red herring. The memcached errors I was getting was due to neutron services not starting, due to an error in the openstack_user_config.yml file. Here is the error from the logs:

2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan [-] Failed to parse network_vlan_ranges. Service terminated!
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan Traceback (most recent call last):
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan   File "/openstack/venvs/neutron-12.0.10/lib/python2.7/site-packages/neutron/plugins/ml2/drivers/type_vlan.py", line 92, in _parse_network_vlan_ranges
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan     cfg.CONF.ml2_type_vlan.network_vlan_ranges)
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan   File "/openstack/venvs/neutron-12.0.10/lib/python2.7/site-packages/neutron/plugins/common/utils.py", line 95, in parse_network_vlan_ranges
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan     network, vlan_range = parse_network_vlan_range(entry)
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan   File "/openstack/venvs/neutron-12.0.10/lib/python2.7/site-packages/neutron/plugins/common/utils.py", line 82, in parse_network_vlan_range
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan     raise n_exc.NetworkVlanRangeError(vlan_range=entry, error=ex)
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan NetworkVlanRangeError: Invalid network VLAN range: 'vlan:1555' - 'need more than 2 values to unpack'
2016-04-19 20:13:10.025 2250 ERROR neutron.plugins.ml2.drivers.type_vlan

Since this customer only had one tenant vlan, I mistakenly entered only one value as the range variable, not realizing it needed to be in the format "vlan1:vlan1". Once corrected, running the play again was successful. Again, my apologies for the post, and thanks for all your hard work.
